### PR TITLE
Adds support for link to account login

### PR DESCRIPTION
**Pull request by axosoft-ramint**
Adds support for links of the format: `vscode://eamodio.gitlens/login?code={code}(&state={state})`

These will login to the GK account using code-token exchange with the provided code/state.

### CODE REVIEW

File: src/plus/gk/account/subscriptionService.ts [hunk 1]
```diff
@@ -1349,6 +1383,18 @@ export class SubscriptionService implements Disposable {
 		);
 	}
 
+	onLoginUri(uri: Uri) {
+		const queryParams: URLSearchParams = new URLSearchParams(uri.query);
+		const code = queryParams.get('code');
+		const state = queryParams.get('state');
+		if (code == null) {
+			void window.showErrorMessage('Unable to login to GitKraken: invalid link.');
```
Comment (eamodio) [inline src/plus/gk/account/subscriptionService.ts:1391]:
```suggestion
			void window.showErrorMessage(`Unable to sign in to GitKraken with that link. Please try clicking that link again. If this issue persists, please contact support.`);
```

File: src/plus/gk/account/authenticationProvider.ts [hunk 2]
```diff
@@ -28,6 +28,12 @@ export const authenticationProviderId = 'gitlens+';
 export const authenticationProviderScopes = ['gitlens'];
 const authenticationLabel = 'GitKraken: GitLens';
 
+export interface AuthenticationProviderOptions {
+	signUp?: boolean;
+	code?: string;
+	state?: string;
```
Comment (eamodio) [inline src/plus/gk/account/authenticationProvider.ts:34]: Combine into a `signIn?: { code: string; state?: string }` prop

File: src/plus/gk/account/subscriptionService.ts [hunk 3]
```diff
@@ -359,14 +361,41 @@ export class SubscriptionService implements Disposable {
 			);
 		}
 
+		return this.loginCore({ signUp: signUp, source: source });
+	}
+
+	async loginWithCode(code: string, state?: string, source?: Source): Promise<boolean> {
```
Comment (eamodio) [inline src/plus/gk/account/subscriptionService.ts:367]: Combine `code` & `state` into an `authentication?: { code: string; state?: string }` argument (I can't think of a better name for the arg as `signIn` like the others doesn't really make sense here)

File: src/plus/gk/account/subscriptionService.ts [hunk 4]
```diff
@@ -359,14 +361,41 @@ export class SubscriptionService implements Disposable {
...
+		}
+
+		return this.loginCore({ code: code, state: state, source: source });
+	}
+
+	private async loginCore(options?: {
+		signUp?: boolean;
+		source?: Source;
+		code?: string;
+		state?: string;
```
Comment (eamodio) [inline src/plus/gk/account/subscriptionService.ts:385]: Combine code & state into a `signIn?: { code: string; state?: string }` argument

**Review by eamodio — CHANGES_REQUESTED** [review]
Overall this looks good. 

Lets allow for a `reason` or `context` querystring (align with gk.dev naming) to allow for better customization of messaging -- e.g. if this is a start trial link flow, we can adjust messaging (and maybe error messaging) accordingly

File: src/plus/gk/account/subscriptionService.ts [hunk 5]
```diff
@@ -1388,11 +1385,13 @@ export class SubscriptionService implements Disposable {
 		const code = queryParams.get('code');
 		const state = queryParams.get('state');
 		if (code == null) {
-			void window.showErrorMessage('Unable to login to GitKraken: invalid link.');
+			void window.showErrorMessage(
+				'Unable to sign in to GitKraken with that link. Please try clicking that link again. If this issue persists, please contact support.',
```
Comment (eamodio) [inline src/plus/gk/account/subscriptionService.ts:1389]: nit:
```suggestion
				'Unable to sign in to GitKraken with that link. Please try clicking the link again. If this issue persists, please contact support.',
```

**Comment by axosoft-ramint** [comment]
@eamodio I added support for a `context` param, but so far all we do with it is customize wording on the top-level error notification. There's probably more we can do - let me know if you can spot any right away.

**Review by eamodio — APPROVED** [review]

**Pull request merged**